ui/src/views/compute/EditVM.vue

```diff
@@ -215,6 +227,9 @@ export default {
           params.haenable = values.haenable
         }
         params.group = values.group
+        if (values.userdata && values.userdata.length > 0) {
+          params.userdata = encodeURIComponent(btoa(this.sanitizeReverse(values.userdata)))
+        }
         this.loading = true
 
         api('updateVirtualMachine', params).then(json => {
```

Comment:
```suggestion
        api('updateVirtualMachine', {}, 'POST', params).then(json => {
```

Would be a good idea to make the API call via POST since the userdata can get quite large

Reply:
@davidjumani 
good point. thanks !